core/system/inc/page_allocator_faults.h

```diff
@@ -30,16 +30,36 @@ uint32_t page_allocator_get_faults(uint8_t page);
 
 /** Map an address to the start and end addresses of a page.
  * If the address is not part of any page, or the page does not belong to the
- * active box or box 0 an error is returned and `start` and `end` are invalid.
+ * active box or box 0 an error is returned and `start`, `end` and `page` are invalid.
  *
- * @param address           the address to map
- * @param[out] start_addr   pointer to the start address value
- * @param[out] end_addr     pointer to the end address value
- * @param[out] page         pointer to the page index value
+ * @param address           the address to locate in the page heap
+ * @param[out] start_addr   the start address of the found page
+ * @param[out] end_addr     the end address of the found page
+ * @param[out] page         the physical page index of the found page
  * @returns Non-zero on failure with failure class `UVISOR_ERROR_CLASS_PAGE`. See `UVISOR_ERROR_PAGE_*`.
  */
 int page_allocator_get_active_region_for_address(uint32_t address, uint32_t * start_addr, uint32_t * end_addr, uint8_t * page);
 
+/** Map an address to an 8-bit page mask.
+ * If the address is not part of any page, or the page does not belong to the
+ * active box or box 0 an error is returned and `mask`, `index` and `page` are invalid.
+ *
+ * Example with 12 pages:
+ *  - right-aligned map: 0b00000000'00000000'00001111'11111111
```

Comment:
In general, is there really a strong case about having 2 implementations for the right- and left-aligned cases? It might a premature optimization that we don't need. What's the runtime difference?